Library/Homebrew/livecheck/strategy.rb

```diff
@@ -181,18 +181,25 @@ def from_url(url, livecheck_strategy: nil, url_provided: false, regex_provided:
       # collected into an array of hashes.
       #
       # @param url [String] the URL to fetch
-      # @param homebrew_curl [Boolean] whether to use brewed curl with the URL
+      # @param options [Hash] options to control behavior
       # @return [Array]
-      sig { params(url: String, homebrew_curl: T::Boolean).returns(T::Array[T::Hash[String, String]]) }
-      def self.page_headers(url, homebrew_curl: false)
+      sig { params(url: String, options: T::Hash[Symbol, T.untyped]).returns(T::Array[T::Hash[String, String]]) }
+      def self.page_headers(url, options = {})
         headers = []
 
+        args = if options[:curl_args].is_a?(Array)
+          PAGE_HEADERS_CURL_ARGS + options[:curl_args]
+        else
+          PAGE_HEADERS_CURL_ARGS
+        end
+
         [:default, :browser].each do |user_agent|
           output, _, status = curl_with_workarounds(
-            *PAGE_HEADERS_CURL_ARGS, url,
+            *args,
+            url,
             **DEFAULT_CURL_OPTIONS,
-            use_homebrew_curl: homebrew_curl,
-            user_agent:        user_agent
+            use_homebrew_curl: options[:homebrew_curl] || false,
+            user_agent:        user_agent,
```

Comment:
Could you refactor this (including `parse_curl_output`) into a `curl_head` method so it can also be reused in `Library/Homebrew/download_strategy.rb`?

Reply:
And in the `curl_head` method, retry with `--request GET` if we don't get a `Content-Disposition` header with `--head` alone.

Reply:
I think that should cover most cases that are broken.

Reply:
I can look into it later tonight but, just off off the top of my head, I would want a `curl_head` method to have a configurable `retry` argument to disable that logic.

As mentioned above, I don't want this retry logic to exist in livecheck once the `options` feature allows us to resolve the issue in `livecheck` blocks. At that point, we want to surface these failures (so we can address them) and keeping the retry logic would lead to us making unnecessary requests forever (since the check would appear to be working fine).

In the interim time, I've updated the list of broken checks in the PR description to include more information (there's a good amount of variety).

Reply:
> keeping the retry logic would lead to us making unnecessary requests forever

We will still need that logic forever to resolve the filename for downloads though.

Reply:
> we want to surface these failures

I don't agree in this particular case. If a URL redirects to a versioned download, the livecheck should just work without any alterations, even if a retry with `--request GET` is required.